internal/zero/bootstrap/writers/writers.go

```diff
@@ -0,0 +1,43 @@
+package writers
+
+import (
+	"context"
+	"crypto/cipher"
+	"fmt"
+	"net/url"
+	"sync"
+
+	cluster_api "github.com/pomerium/pomerium/pkg/zero/cluster"
+)
+
+type ConfigWriter interface {
+	WriteConfig(ctx context.Context, src *cluster_api.BootstrapConfig, cipher cipher.AEAD) error
```

Comment:
I'm not sure about passing cipher in this function:
- if we keep encryption for both writers, lets make encrypt/decrypt responsibility of the caller. 
- if we only keep encryption for the file writer, the cipher need be provided to it during initialization

Reply:
Yeah I agree, it makes more sense to keep encryption as an implementation detail.